File: ROS2/Day1/Day1_ROS2_package/changgyu_task2_py/changgyu_task2_py/subscriber.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import Int32
from std_msgs.msg import String
from std_msgs.msg import Float32
from std_msgs.msg import Bool


class Task2Subscriber(Node):

    def __init__(self):
        super().__init__('task2_py_subscriber')

        self.int_subscriber = self.create_subscription(
            Int32,
            '/task2/int_data',
            self.int_callback,
            10
        )

        self.string_subscriber = self.create_subscription(
            String,
            '/task2/string_data',
            self.string_callback,
            10
        )

        self.float_subscriber = self.create_subscription(
            Float32,
            '/task2/float_data',
            self.float_callback,
            10
        )

        self.bool_subscriber = self.create_subscription(
            Bool,
            '/task2/bool_data',
            self.bool_callback,
            10
        )

    def int_callback(self, msg):
        self.get_logger().info(
            'Received int: {0}'.format(msg.data)
        )

    def string_callback(self, msg):
        self.get_logger().info(
            'Received string: {0}'.format(msg.data)
        )

    def float_callback(self, msg):
        self.get_logger().info(
            'Received float: {0:.2f}'.format(msg.data)
        )

    def bool_callback(self, msg):
        self.get_logger().info(
            'Received bool: {0}'.format(msg.data)
        )


def main(args=None):
    rclpy.init(args=args)

    node = Task2Subscriber()

    try:
        rclpy.spin(node)

    except KeyboardInterrupt:
        node.get_logger().info('Keyboard Interrupt (SIGINT)')

    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()

File: ROS2/Day1/Day1_ROS2_package/changgyu_task2_cpp/src/subscriber.cpp
#include "changgyu_task2_cpp/subscriber.hpp"

#include <functional>

using std::placeholders::_1;

Task2Subscriber::Task2Subscriber()
    : Node("task2_cpp_subscriber")
{
    int_subscriber_ =
        this->create_subscription<std_msgs::msg::Int32>(
            "/task2/int_data",
            10,
            std::bind(&Task2Subscriber::int_callback, this, _1));

    string_subscriber_ =
        this->create_subscription<std_msgs::msg::String>(
            "/task2/string_data",
            10,
            std::bind(&Task2Subscriber::string_callback, this, _1));

    float_subscriber_ =
        this->create_subscription<std_msgs::msg::Float32>(
            "/task2/float_data",
            10,
            std::bind(&Task2Subscriber::float_callback, this, _1));

    bool_subscriber_ =
        this->create_subscription<std_msgs::msg::Bool>(
            "/task2/bool_data",
            10,
            std::bind(&Task2Subscriber::bool_callback, this, _1));
}

void Task2Subscriber::int_callback(
    const std_msgs::msg::Int32::SharedPtr msg)
{
    RCLCPP_INFO(
        this->get_logger(),
        "Received int: %d",
        msg->data);
}

void Task2Subscriber::string_callback(
    const std_msgs::msg::String::SharedPtr msg)
{
    RCLCPP_INFO(
        this->get_logger(),
        "Received string: %s",
        msg->data.c_str());
}

void Task2Subscriber::float_callback(
    const std_msgs::msg::Float32::SharedPtr msg)
{
    RCLCPP_INFO(
        this->get_logger(),
        "Received float: %.2f",
        msg->data);
}

void Task2Subscriber::bool_callback(
    const std_msgs::msg::Bool::SharedPtr msg)
{
    RCLCPP_INFO(
        this->get_logger(),
        "Received bool: %s",
        msg->data ? "true" : "false");
}

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<Task2Subscriber>();
    rclcpp::spin(node);

    rclcpp::shutdown();

    return 0;
}